### PR TITLE
Remove deprecated methods in AggregationNode

### LINKED ISSUE #7922: Remove the deprecated methods in AggregationNode

### PR DESCRIPTION
**Pull request by rongrong**
Resolves #7922

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java [hunk 1]
```diff
@@ -1892,13 +1893,13 @@ private PhysicalOperation planGlobalAggregation(int operatorId, AggregationNode
...
                 Symbol symbol = entry.getKey();
-
+                Aggregation aggregation = entry.getValue();
                 accumulatorFactories.add(buildAccumulatorFactory(source,
-                        node.getFunctions().get(symbol),
-                        entry.getValue(),
-                        node.getMasks().get(entry.getKey())));
+                        aggregation.getSignature(),
+                        aggregation.getCall(),
+                        aggregation.getMask().orElse(null)));
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java]: Maybe `buildAccumulatorFactory` should accept `Optional mask` instead?

File: presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java [hunk 2]
```diff
@@ -519,8 +519,9 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         aggregationTranslations.copyMappingsFrom(groupingTranslations);
 
         // 2.d. Rewrite aggregates
-        ImmutableMap.Builder<Symbol, FunctionCall> aggregationAssignments = ImmutableMap.builder();
-        ImmutableMap.Builder<Symbol, Signature> functions = ImmutableMap.builder();
+        ImmutableMap.Builder<Symbol, Aggregation> aggregations = ImmutableMap.builder();
+        // Map from aggregate function arguments to marker symbols, so that we can reuse the markers, if two aggregates have the same argument
+        Map<Set<Expression>, Symbol> argumentMarkers = new HashMap<>();
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java]: can Expression be `random()`? Is reusing safe?

Comment (rongrong) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java]: Could you please explain it a bit more? Thanks!

Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java]: I see now you didn't introduce the map (refactoring only), so maybe that's not that important

File: presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java [hunk 3]
```diff
@@ -519,8 +519,9 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         aggregationTranslations.copyMappingsFrom(groupingTranslations);
 
         // 2.d. Rewrite aggregates
-        ImmutableMap.Builder<Symbol, FunctionCall> aggregationAssignments = ImmutableMap.builder();
-        ImmutableMap.Builder<Symbol, Signature> functions = ImmutableMap.builder();
+        ImmutableMap.Builder<Symbol, Aggregation> aggregations = ImmutableMap.builder();
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java]: Move these closer to where they're used

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeLimitWithDistinct.java [hunk 4]
```diff
@@ -62,7 +62,7 @@
      */
     private boolean isDistinct(AggregationNode node)
     {
-        return !node.getAggregations().isEmpty() ||
+        return !node.getAssignments().isEmpty() ||
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeLimitWithDistinct.java]: I thought this diff was about moving to Aggregation not away from it.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SingleMarkDistinctToGroupBy.java [hunk 5]
```diff
@@ -75,12 +80,15 @@
         // optimize if and only if
         // all aggregation functions have a single common distinct mask symbol
         // AND all aggregation functions have mask
-        Set<Symbol> masks = ImmutableSet.copyOf(parent.getMasks().values());
-        if (masks.size() != 1 || parent.getMasks().size() != parent.getAggregations().size()) {
+        Collection<Aggregation> aggregations = parent.getAssignments().values();
+        List<Symbol> masks = aggregations.stream()
+                .map(Aggregation::getMask).filter(Optional::isPresent).map(Optional::get).collect(toImmutableList());
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SingleMarkDistinctToGroupBy.java]: Always separate multiple function calls like this with newlines.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java (hunk 3 above)
Comment (rongrong) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java]: It's used in the for loop.

**Pull request closed without merging**